app/vminsert/clusternative/request_handler.go

```diff
@@ -25,7 +26,7 @@ var (
 func InsertHandler(c net.Conn) error {
 	// There is no need in response compression, since
 	// lower-level vminsert sends only small packets to upper-level vminsert.
-	bc, err := handshake.VMInsertServer(c, 0)
+	bc, err := handshake.VMInsertServer(c, 0, netstorage.GetNodeID())
```

Comment:
this hash is not persistent. Each update to underlying storage nodes breaks it.

So it makes no sense to return it.

Reply:
> this hash is not persistent. Each update to underlying storage nodes breaks it.

Yes, this hash is not persisted, but it is expected that underlying storage node IDs are persisted, so building a hash based on these IDs should return the same result. Could you elaborate on each update to storage nodes forcing to change the ID?